src/apps/material/helper.ts

```diff
@@ -454,6 +457,35 @@ export const useGetHoldings = ({
   return { data, isLoading, isError };
 };
 
+export const getManifestationBasedOnType = (
+  work: Work,
+  materialType: DisplayMaterialType
+): Manifestation => {
+  const { bestRepresentation, all } = work.manifestations;
+
+  // Retrieves the material type of the best representation.
+  const bestRepresentationMaterialType =
+    getManifestationMaterialTypes(bestRepresentation);
+
+  // Checks if the best representation matches the desired material type.
+  if (materialType === bestRepresentationMaterialType) {
+    return bestRepresentation;
+  }
+
+  // Filters and sorts the manifestations if the best representation does not match.
+  const filteredByType = filterManifestationsByType(materialType, all);
+  const sortedByYear = orderManifestationsByYear(filteredByType);
```

Comment:
Consider dropping the explicit ordering.


`filterManifestationsByType()` already does ordering by year (even though it is not evident by the title). You replace the ordering by a comment.